### PR TITLE
[IMP] amazon: migrate to SP-API

### PR DESCRIPTION
**Pull request by Demesmaeker**
Starting July 2022, Amazon WMS API is deprecated and we need to use the Selling Partners API. 

This brings changes in the authentication flow that are explained here.

task-2466636

See also:
- https://github.com/odoo/enterprise/pull/18597
- https://github.com/odoo/iap-apps/pull/419

### CODE REVIEW

File: content/applications/sales/sales/amazon_connector/setup.rst
Comment (meng-odoo) [inline content/applications/sales/sales/amazon_connector/setup.rst:19]: Changed wording to make it flow better.

`confirm that you want Amazon to give Odoo access to your account and related data.`

**Review by meng-odoo — CHANGES_REQUESTED** [review]
@Demesmaeker Hi! Thanks for updating this doc. I'm just requesting some minor corrections before we approve :)

**Comment by Demesmaeker** [comment]
Hello @meng-odoo I made the changes you requested. Also, I forgot to make the changes regarding the creation of a new marketplace, it's now done.

Thanks for your feedback, and here is a small tip for your reviews: you can `make a suggestion`, this is easier both for you (you don't have to rewrite a sentence) and for me (as I'd immediately see the changes).

image

Have a nice day =]

**Review by meng-odoo — CHANGES_REQUESTED** [review]
Hello @Demesmaeker! Thank you very much for making those changes. Also, thanks for the tip :) I am new to using GitHub so I'm still learning!

Upon a second review, I think we should also fix a couple formatting details, as well as delete some instances of second person familiar ("you can") writing in order to maintain a strong instructional voice throughout the doc. Please see my comments below and let me know if you have any questions.

**Comment by Demesmaeker** [comment]
Hello @meng-odoo 
Changes are done, I had to specify my label name for the 2 last links as I couldn't have the same label for both, and I also added some changes in the `update.rst`, which I totally forgot about before.

File: content/applications/sales/sales/amazon_connector/update.rst
Comment (meng-odoo) [inline content/applications/sales/sales/amazon_connector/update.rst:8]: Let's delete this line. Normally we do not preface the documentation by saying what information will be included in each one.

File: content/applications/sales/sales/amazon_connector/update.rst
Comment (meng-odoo) [inline content/applications/sales/sales/amazon_connector/update.rst:11]: Reworking this sentence for present tense and flow.
```suggestion
update module is installed automatically. To check if this module is already installed,
```

File: content/applications/sales/sales/amazon_connector/update.rst
Comment (meng-odoo) [inline content/applications/sales/sales/amazon_connector/update.rst:12]: Reworking this sentence (continued from the last comment) and fixing some Odoo terminology.
```suggestion
go to :guilabel:`Apps`, remove the ``Apps`` filter and search for ``Amazon``.
```

File: content/applications/sales/sales/amazon_connector/update.rst
Comment (meng-odoo) [inline content/applications/sales/sales/amazon_connector/update.rst:10]: Let's turn this into a Note to grab the reader's attention and separate this information from the rest of the instructions on the doc. Also, I'm editing the tense to keep it consistent throughout the sentence.
```suggestion
.. note::
If your Odoo database is created after the update was released, the
```

File: content/applications/sales/sales/amazon_connector/update.rst
Comment (meng-odoo) [inline content/applications/sales/sales/amazon_connector/update.rst:48]: Editing wording and punctuation.
```suggestion
    you should use a staging environment--contact your integrating partner for
```

File: content/applications/sales/sales/amazon_connector/update.rst
Comment (meng-odoo) [inline content/applications/sales/sales/amazon_connector/update.rst:53]: Fixing some Odoo terminology.
```suggestion
The module should now be available in your :guilabel:`Apps`. Remove the ``Apps`` filter
and search for ``Amazon``; the module :guilabel:`Amazon/Selling Partner API Patch` should be
available for installation. If you cannot find the module after having updated the list of available
```

File: content/applications/sales/sales/amazon_connector/update.rst
Comment (meng-odoo) [inline content/applications/sales/sales/amazon_connector/update.rst:58]: Editing for brevity and instructional voice.
```suggestion
Once the module is installed, link your Amazon account by following the instructions on the :doc:`setup <setup>` page.
```

**Review by meng-odoo — CHANGES_REQUESTED** [review]
Hi @Demesmaeker! I'm just requesting a few more tiny changes on the Setup doc, and some small wording changes on the Update doc to maintain instructional voice, grammar, and consistent terminology. Please let me know if you have any questions. Thanks!

File: content/applications/sales/sales/amazon_connector/update.rst
Comment (Demesmaeker) [reply content/applications/sales/sales/amazon_connector/update.rst:48]: I understand why we should take off `you should` here, but I'm not too sure about the replacement, is it something I'm not english-speakery enough to know about? :sweat_smile:  I'd have used `:` instead, I think. I put your suggestion anyway, but keep me updated so that I can change it if necessary (or learned an interesting english uses of the punctuation :smile: ). 

(BTW, since we are changing things in this paragraph, I fixed the indentation as well, so don't mind the huge diff, I only changed what you told me to ^^)

File: content/applications/sales/sales/amazon_connector/update.rst
Comment (meng-odoo) [inline content/applications/sales/sales/amazon_connector/update.rst:43]: Adding a word for clarity.
```suggestion
   You should never install new modules in your production database without first testing them in a
```

File: content/applications/sales/sales/amazon_connector/update.rst
Comment (meng-odoo) [inline content/applications/sales/sales/amazon_connector/update.rst:46]: Let's fix this punctuation either with a semicolon `;` or an em dash `---` as we discussed :)
```suggestion
   database. For on-premise users, you should use a staging environment---contact your integrating
```

**Review by meng-odoo — CHANGES_REQUESTED** [review]
Hi @Demesmaeker! Thanks for making those changes. Everything looks great, we just need to fix the punctuation on that one part that you asked about, and I requested one more tiny wording change, since we're already modifying. Please see my comments below and let me know if you have any questions :)

File: content/applications/sales/sales/amazon_connector/update.rst
Comment (Demesmaeker) [reply content/applications/sales/sales/amazon_connector/update.rst]: Ho, nice, indeed, `---` did the trick :partying_face:

**Review by meng-odoo — APPROVED** [review]
@Demesmaeker Thanks for making those changes! I'm glad the punctuation worked :) Very good to know! Everything looks good now. I'll go ahead and move this PR to the next stage in the review process.

File: content/applications/sales/sales/amazon_connector/update.rst
Comment (AntoineVDV) [inline content/applications/sales/sales/amazon_connector/update.rst:2]: Shouldn't we change the title as well?

File: content/applications/sales/sales/amazon_connector/update.rst
Comment (AntoineVDV) [inline content/applications/sales/sales/amazon_connector/update.rst:40]: Same here :point_up:

File: content/applications/sales/sales/amazon_connector/setup.rst
Comment (AntoineVDV) [inline content/applications/sales/sales/amazon_connector/setup.rst:27]: I wouldn't advertise that (or at least not in the "register your account section), especially since it's only available in debug mode.

File: content/applications/sales/sales/amazon_connector/setup.rst
Comment (AntoineVDV) [inline content/applications/sales/sales/amazon_connector/setup.rst:80]: I believe it's important to state that those marketplaces at not supported. Maybe combine both statements?

File: content/applications/sales/sales/amazon_connector/setup.rst
Comment (AntoineVDV) [inline content/applications/sales/sales/amazon_connector/setup.rst:156]: Same here.

**Review by AntoineVDV — CHANGES_REQUESTED** [review]

File: content/applications/sales/sales/amazon_connector/setup.rst
Comment (Demesmaeker) [reply content/applications/sales/sales/amazon_connector/setup.rst:80]: They should work, that's why on the application in the Seller Central, we put that every marketplace are supported, @dalaOdoo has a Japanese contact who will test it I think (well, we talked about it at least, I don't know if the customer is still interested). If it works in Japan, it'll work everywhere, and we can say they are supported.

File: content/applications/sales/sales/amazon_connector/setup.rst
Comment (Demesmaeker) [reply content/applications/sales/sales/amazon_connector/setup.rst:156]: I'm not sure... :thinking: This should work whatever the marketplace is, the SP-API is more consistent for its variable names than MWS was

File: content/applications/sales/sales/amazon_connector/setup.rst
Comment (Demesmaeker) [reply content/applications/sales/sales/amazon_connector/setup.rst:27]: I delete it, I feel it doesn't have it's place in the `manage` file either. But I can add it there if you wish =]

File: content/applications/sales/sales/amazon_connector/setup.rst
Comment (Demesmaeker) [reply content/applications/sales/sales/amazon_connector/setup.rst:78]: After discussing it, this'll be changed in master

File: content/applications/sales/sales/amazon_connector/setup.rst
Comment (Demesmaeker) [reply content/applications/sales/sales/amazon_connector/setup.rst:156]: After discussing it, this'll be changed in master

**Review by AntoineVDV — APPROVED** [review]
@robodoo delegate+

**Comment by AntoineVDV** [comment]
@robodoo r+

**Pull request closed without merging**
